R/get_ref.R

```diff
@@ -414,11 +420,64 @@ get_release_date.remote_ref <- function(remote_ref) {
   NA
 }
 
+#' Get the pseudo-published date of Bioconductor packages
+#'
+#' The date from the last modification to the sources is assumed to be the
+#' published date as it is not available in `pkgcache` results.
+#'
+#' The official metadata of the Bioconductor release does not work for the
+#' purpose of this package (`https://bioconductor.org/config.yaml`). As there
+#' might be a edge case where a Bioconductor package was changed and depends
+#' on a newer version of a package. (Actual edge case with `SummarizedExperiment`
+#' and `matrixStats`)
+#'
+#' @keywords internal
+get_bioc_package_release_date <- function(package) {
+  pkg_list <- pkgcache::meta_cache_list(packages = package)
+  tryCatch(
+    {
+      pkg_list <- pkg_list[order(package_version(pkg_list$version)), ][1, ]
+      source_url <- pkg_list$sources[[1]]
+      last_modified <- httr::GET(source_url)$headers$`last-modified`
+      as.POSIXct(strptime(last_modified, format = "%a, %d %b %Y %H:%M:%S"))
+    },
+    error = function(err) {
+      cli::cli_alert_info(paste0(
+        "Couldn't find the release date for ",
+        cli::col_blue("bioc::{{{package}}}"),
+        ". This may cause a failure to install packages as",
+        " min_cohort / min_isolated strategies depend on the release date."
+      ))
+      NA
+    }
+  )
+}
```

Comment:
Problem with Bioconductor versions as they don't have any metadata information about the publication date.

```R
r> pkgcache::meta_cache_list("SummarizedExperiment")$published
[1] NA
```

**This is very open to change!!**

Workaround checks the last modified date of the package source.

However, I couldn't find an API that would retrieve the information and using Bioconductor release date doesn't work as specific edge case has 2 packages with new releases _(where 1 depends on the other)_.

Reply:
Yeah, packages from Bioconductor are little bit more difficult to handle and I don't have a good solution for that. I just did fallback into `remote_ref` class and return NA and then handle NAs in the following codes.

Reply:
There is a (bioconductor) package called BiocPkgTools ([GH page](github.com/seandavi/BiocPkgTools)) that fetches some metadata. I have read it's source code and it seems that they are parsing URL from the current bioc version (obtained via `BiocManager::version()`) into something like this: https://bioconductor.org/packages/3.17/bioc/VIEWS
They then open url connection, read that file (`read.dcf()`) and reports everything including `"Date/Publication"` field.

Reply:
Please have a look at this: https://code.bioconductor.org/browse/hermes/commits/devel

It looks like Bioc is doing artificial vbump even if package has not changed since the last release resulting in artificial update of the last release date. This algorithm cannot handle Bioc packages in the same way as we handle CRAN and GH packages.

Reply:
As we discussed removed the custom to check for the timestamp.

Using the current date as a placeholder for Bioc packages as it fails with `SummarizedExperiment` or do you want for it to fail completely? _(I believe it's used on `teal.data`)_

Created issue upstream on https://github.com/r-lib/pkgcache/issues/99